crates/biome_cli/src/commands/format.rs

```diff
@@ -72,64 +74,91 @@ pub(crate) fn format(
+            console.error(markup! {
+                {PrintDiagnostic::simple(&diagnostic)}
+            });
+
+            js_formatter.indent_width = Some(indent_size);
+        }
     }
     // TODO: remove in biome 2.0
-    if json_formatter
-        .as_ref()
-        .is_some_and(|f| f.indent_size.is_some())
-    {
-        let console = &mut session.app.console;
-        let diagnostic = DeprecatedArgument::new(markup! {
-            "The argument "<Emphasis>"--json-formatter-indent-size"</Emphasis>" is deprecated, it will be removed in the next major release. Use "<Emphasis>"--json-formatter-indent-width"</Emphasis>" instead."
-        });
-        console.error(markup! {
-            {PrintDiagnostic::simple(&diagnostic)}
-        })
+    if let Some(json_formatter) = json_formatter.as_mut() {
+        if let Some(indent_size) = json_formatter.indent_size {
+            let console = &mut session.app.console;
+            let diagnostic = DeprecatedArgument::new(markup! {
+                "The argument "<Emphasis>"--json-formatter-indent-size"</Emphasis>" is deprecated, it will be removed in the next major release. Use "<Emphasis>"--json-formatter-indent-width"</Emphasis>" instead."
+            });
+            console.error(markup! {
+                {PrintDiagnostic::simple(&diagnostic)}
+            });
+
+            json_formatter.indent_width = Some(indent_size);
+        }
     }
     // TODO: remove in biome 2.0
-    if css_formatter
-        .as_ref()
-        .is_some_and(|f| f.indent_size.is_some())
-    {
-        let console = &mut session.app.console;
-        let diagnostic = DeprecatedArgument::new(markup! {
-            "The argument "<Emphasis>"--css-formatter-indent-size"</Emphasis>" is deprecated, it will be removed in the next major release. Use "<Emphasis>"--css-formatter-indent-width"</Emphasis>" instead."
-        });
-        console.error(markup! {
-            {PrintDiagnostic::simple(&diagnostic)}
-        })
+    if let Some(css_formatter) = css_formatter.as_mut() {
+        if let Some(indent_size) = css_formatter.indent_size {
+            let console = &mut session.app.console;
+            let diagnostic = DeprecatedArgument::new(markup! {
+                "The argument "<Emphasis>"--css-formatter-indent-size"</Emphasis>" is deprecated, it will be removed in the next major release. Use "<Emphasis>"--css-formatter-indent-width"</Emphasis>" instead."
+            });
+            console.error(markup! {
+                {PrintDiagnostic::simple(&diagnostic)}
+            });
+
+            css_formatter.indent_width = Some(indent_size);
+        }
     }
 
-    configuration.merge_with(javascript_formatter);
-    configuration.merge_with(json_formatter);
-    configuration.merge_with(css_formatter);
-    configuration.merge_with(formatter_configuration);
-    configuration.merge_with(vcs_configuration);
-    configuration.merge_with(files_configuration);
+    configuration.merge_with(Configuration {
```

Comment:
I am a bit conflicted about this approach.

We're creating a **new configuration object** that gets disposed of immediately. The former approach didn't create any new intermediate types. Our configuration is already chunky.

Reply:
I'm not very opinionated on this one. I'd be happy to rewrite it so they get merged separately. It might be slightly more verbose, but it wouldn't be a big deal IMO.